src/integer_value.cpp

```diff
@@ -116,27 +117,26 @@ ValuePtr IntegerValue::mul(Env *env, ValuePtr arg) {
     }
     arg.assert_type(env, Value::Type::Integer, "Integer");
 
+    if (is_zero() || arg->as_integer()->is_zero())
```

Comment:
This would cause hydration of the argument if is_zero() is false, right? Maybe we could just check for to_nat_int_t() == 0?

Reply:
Yes, in fact I don't like that so I will be adding a way to check if a ValuePtr is zero and integer without hydration to check, and use that to check arg. Nice call ;) 

PS: the rest of this 0-check will just stay the same and it will do hydration on arg, but only if really needed :)

Reply:
I'm not sure if I understand the "PS" section 😄 Why would we need to hydrate the argument here? In any case we can just check if it's zero by calling to_nat_int_t() on it right? Like `is_zero() || arg.to_nat_int_t() == 0`

Reply:
Oh I see, I see, we can honestly just use to_nat_int_t and totally skip the is_zero since to_nat_int_t in ValuePtr doesn't hydrate and is smart enough to check value() if it isn't an int!